indico/htdocs/js/utils/forms.js

```diff
@@ -64,8 +64,13 @@
         forms.find('[data-disabled-until-change]').prop('disabled', true);
         forms.each(function() {
             var $this = $(this);
-            $this.data('initialData', $this.serialize());
-            $this.data('fieldsChanged', false);
+            function _resetData() {
+                $this.data('initialData', $this.serialize());
+                $this.data('fieldsChanged', false);
+            }
+            _resetData();
```

Comment:
shouldn't triggering `fieldsSaved` result in the submit button being disabled again until further changes? because the logic for that is only in the change/input event handler